### PR TITLE
update CupertinoPicker.selectionOverlay to nullable

### LINKED ISSUE #79196: CupertinoPicker's `selectionOverlay` property is not nullable
[According to the docs](https://api.flutter.dev/flutter/cupertino/CupertinoPicker/selectionOverlay.html) the "property can be set to null to remove the overlay", but CupertinoPicker's `selectionOverlay` is of type `Widget` instead of `Widget?`

## Steps to Reproduce

Create a CupertinoPicker widget and try to set the `selectionOverlay` to `null`. For example:

```
CupertinoPicker(
  selectionOverlay: null,
  backgroundColor: Colors.transparent,
  itemExtent: 100,
  onSelectedItemChanged: (selected) {},
  children: [Text("1"), Text("2"), Text("3")]
),
```

**Expected results:** Be able to set `selectionOverlay` to `null` with no issues.

**Actual results:** 
```
The argument type 'Null' can't be assigned to the parameter type 'Widget'.
```

<details>
  <summary>flutter analyze</summary>

```
Analyzing cupertinopicker_bug...                                        

  error • The argument type 'Null' can't be assigned to the parameter type 'Widget' • lib/main.dart:98:15 • argument_type_not_assignable
```
</details>

<details>
  <summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel dev, 2.1.0-12.1.pre, on macOS 11.2.3 20D91 darwin-x64, locale en-GB)
    • Flutter version 2.1.0-12.1.pre at /Users/tom/flutter
    • Framework revision 8264cb3e8a (2 weeks ago), 2021-03-10 12:37:57 -0800
    • Engine revision 711ab3fda0
    • Dart version 2.13.0 (build 2.13.0-116.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/tom/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    ! CocoaPods 1.9.3 out of date (1.10.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.54.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.20.0
```

</details>

### PR DESCRIPTION
**Pull request by xu-baolin**
Fixes #79196

### CODE REVIEW

File: packages/flutter/lib/src/cupertino/picker.dart [hunk 1]
```diff
@@ -259,7 +259,7 @@ class _CupertinoPickerState extends State<CupertinoPicker> {
   }
 
   /// Draws the selectionOverlay.
-  Widget _buildSelectionOverlay(Widget selectionOverlay) {
+  Widget _buildSelectionOverlay(Widget? selectionOverlay) {
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/cupertino/picker.dart:262]: nit: I think you can skip building the entire stack entry if `selectionOverlay` is null, in the `build` method.

**Review by LongCatIsLooong — APPROVED** [review]
Thanks for the fix! \cc @YeungKC in case you want to take a look.

File: packages/flutter/lib/src/cupertino/picker.dart (hunk 1 above)
Comment (YeungKC) [reply packages/flutter/lib/src/cupertino/picker.dart:262]:
```dart

if (widget.selectionOverlay != null) {
  _buildSelectionOverlay(widget.selectionOverlay)
}

```

I think so too.

**Review by YeungKC — COMMENTED** [review]
LGTM, but I think the same as @LongCatIsLooong.

**Pull request merged**